src/service/OAuthService.ts

```diff
@@ -0,0 +1,74 @@
+import { Request, Response, Router } from "express"
+import { OAuthConfiguration } from "../utils/OAuthConfiguration"
+import { _verify } from "./Security"
+import fetch, { Request as Req, Response as Res } from "node-fetch"
+import { CredentialService } from './CredentialService';
+import { decode } from "jsonwebtoken"
+
+export class OAuthService {
+  public static _name = "OAuth"
+  public static Router = Router()
+}
+
+OAuthService.Router.get("/oauth/start", async (req: Request, res: Response) => {
+  if (!OAuthConfiguration.isEnabled) {
+    res
+      .status(404)
+      .json({ message: "404.oauth-disabled" })
+    return
+  }
+
+  let loginURL = OAuthConfiguration.getLoginURL()
+  if (!loginURL) {
+    res.status(500).send("Internal server errror")
+    return
+  }
+
+  let logoutURL = OAuthConfiguration.getLogoutURL()
+
+  res.json({
+    loginURL,
+    logoutURL,
+  })
+})
+
+OAuthService.Router.post("/oauth/authenticate", async (req: Request, res: Response) => {
```

Comment:
Is this method only called by the IdP? Or also by the API client?

Reply:
In the future we want to bring part of this to the backend, but right now it's sort of complicated.